internal/devbox/packages.go

```diff
@@ -571,3 +419,93 @@ func (d *Devbox) InstallRunXPackages(ctx context.Context) error {
 	}
 	return nil
 }
+
+// installNixPackagesToStore will install all the packages in the nix store, if
+// mode is install or update, and we're not in a devbox environment.
+// This is done by running `nix build` on the flake. We do this so that the
+// packages will be available in the nix store when computing the devbox environment
+// and installing in the nix profile (even if offline).
+func (d *Devbox) installNixPackagesToStore(ctx context.Context) error {
+	packages, err := d.packagesToInstallInProfile(ctx)
+	if err != nil {
+		return err
+	}
+
+	stepNum := 0
+	total := len(packages)
+	for _, pkg := range packages {
+		stepNum += 1
+
+		installable, err := pkg.Installable()
+		if err != nil {
+			return err
+		}
+
+		stepMsg := fmt.Sprintf("[%d/%d] %s", stepNum, total, pkg)
+		fmt.Fprintf(d.stderr, stepMsg+"\n")
+
+		// --no-link to avoid generating the result objects
+		err = nix.Build(ctx, []string{"--no-link"}, installable)
+		if err != nil {
+			fmt.Fprintf(d.stderr, "%s: ", stepMsg)
+			color.New(color.FgRed).Fprintf(d.stderr, "Fail\n")
+
+			platform := nix.System()
+			return usererr.WithUserMessage(
```

Comment:
I can tighten this condition a bit, but its hard to make it as good as before.

Previously, we could check `strings.Contains(string(out), "is not available on the requested hostPlatform") {`

but now the error looks like:
```
source: nix build exit code: 1, 
output: error: 
flake output attribute 'legacyPackages.x86_64-darwin.glibcLocales' is not a derivation or path
, err: exit status 1
```

Reply:
With update, it looks like this now:

<img width="1284" alt="Screenshot 2024-01-16 at 10 12 33 AM" src="https://github.com/jetpack-io/devbox/assets/676452/27e520c4-0af6-4924-aa13-9abf67c3accb">